webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/nodes/Properties.scala

```diff
@@ -77,6 +77,62 @@ import net.liftweb.json.JsonDSL._
  */
 final case class PropertyProvider(value: String) extends AnyVal
 
+
+/*
+ * An inherit mode is three char, one for object, one for array, one for string like (?moo)
+ * The first char is for objects and can be: d = o = override, m = merge (default)
+ * The second char is for arrays and can be: d = o = override (default), a = append, p = prepend
+ * The third is for strings and can ve:      d = o = override (default), a = append, p = prepend
+ */
+final case class InheritMode(
+    forObject: InheritMode.ObjectMode
+  , forArray : InheritMode.ArrayMode
+  , forString: InheritMode.StringMode
+) {
+  def value = s"${forObject.value}${forArray.value}${forString.value}"
+}
+final object InheritMode {
+  sealed trait ObjectMode { def value: Char }
+  final case object ObjectMode {
+    final case object Override extends ObjectMode { override val value = 'o' }
+    final case object Merge    extends ObjectMode { override val value = 'm' }
+
+    def all = ca.mrvisser.sealerate.values[ObjectMode].toList
+    def parse(c: Char) = all.find(c == _.value)
+  }
+  sealed trait ArrayMode { def value: Char }
+  final case object ArrayMode {
+    final case object Override extends ArrayMode { override val value = 'o' }
+    final case object Append   extends ArrayMode { override val value = 'a' }
+    final case object Prepend  extends ArrayMode { override val value = 'p' }
+
+    def all = ca.mrvisser.sealerate.values[ArrayMode].toList
+    def parse(c: Char) = all.find(c == _.value)
+  }
+  sealed trait StringMode { def value: Char }
+  final case object StringMode {
+    final case object Override extends StringMode { override val value = 'o' }
+    final case object Append   extends StringMode { override val value = 'a' }
+    final case object Prepend  extends StringMode { override val value = 'p' }
+
+    def all = ca.mrvisser.sealerate.values[StringMode].toList
+    def parse(c: Char) = all.find(c == _.value)
+  }
+
+  def parseString(s: String): Option[InheritMode] = s.toList match {
```

Comment:
I think it should be a PureResult and give errors when parsing is not OK

Maybe also treat the case when there is more then 3 characters ? But maybe it's better to have only 3 chars as valid case, so maybe it's good like this, just discussing

Reply:
I'm not sure. It complicates a lot the usage.
And no, the only legit case if these three chars, so it's ok to return none in other cases